Lib/_pyio.py

```diff
@@ -40,6 +40,28 @@
 _CHECK_ERRORS = _IOBASE_EMITS_UNRAISABLE
 
 
+def text_encoding(encoding, stacklevel=1):
+    """
+    Helper function to choose the text encoding.
+
+    When encoding is not None, just return it.
+    Otherwise, return the default text encoding (i.e. "locale").
+
+    This function emits EncodingWarning if *encoding* is None and
+    sys.flags.warn_default_encoding is true.
+
+    This function can be used in APIs having encoding=None option.
+    But please consider encoding="utf-8" for new APIs.
+    """
+    if encoding is None:
+        if sys.flags.warn_default_encoding:
```

Comment:
You can merge the two if, no?

Reply:
If sys.flags.warn_default_encoding is false, we need to `encoding = "locale"`.

Reply:
Oh right, I didn't notice the indentation of `encoding = "locale"`, you are right :-)